docs/rfcs/1002-merging-process.md

```diff
@@ -0,0 +1,383 @@
+| Feature      | The PR adds a new feature                      |
+| Typo         | The PR fixes a typo                            |
+| Dependencies | The PR updates one or more dependencies        |
+| Release      | The PR is a release                            |
+| RFC          | The PR is related to the RFCs documentation    |
+| Docs         | The PR is related to the user documentation    |
+| Gui          | The PR is related to the GUI                   |
+| Test         | The PR is related to the test codebase         |
+| **PER CODE** |                                                |
+| CI           | The PR affect the CI                           |
+| Rust         | The PR affect the Rust code                    |
+| Python       | The PR affect the Python code                  |
+| Js           | The PR affect the Javascript / Typescript code |
+| Java         | The PR affect the Java / Kotlin / Android code |
+| **PER OS**   |                                                |
+| Linux        | The PR is specific to Linux                    |
+| MacOS        | The PR is specific to MacOS                    |
+| Windows      | The PR is specific to Windows                  |
+
+Hints:
+
+- If the pull request is related to more than one category in `PER CODE`, this
+  may suggests that it needs to be splitted into smaller pull requests.
+  > This would not be enforced because sometime it is required to change
+  > multiple parts of the code at the same time.
+
+- If the pull request is related to more than one category in `PER OS`, this
+  will certainly indicate that the change is not OS-specific so spliting the
+  pull request is recommended.
+
+### The pull request checklist
+
+The checklist is displayed before submitting a pull request:
+
+```markdown
+<!-- You can erase any parts of this template not applicable to your pull request -->
+
+## Describe your changes
+
+<!-- Why this PR exist ? what is its goal ?
+     Give some context about the changes
+     Draw attention to the details that should not be overlooked -->
+
+## Checklist
+
+Before you submit this pull request, please make sure to:
+
+- [ ] Keep changes in the pull request as small as possible
+   <!-- Move unrelated changes to a new pull request -->
+- [ ] Ensure the commit history is sanitized
+   <!-- Commits should have a meaningful title and contain a coherent set of changes
+        Squash commits that are only relevant to the branch context -->
+- [ ] Give a meaningful title to your PR
+- [ ] Describe your changes
+   <!-- Give some context about the changes
+        Draw attention to the details that should not be overlooked -->
+- [ ] Link any related issue in the description
+   <!-- You can add `closes #<IssueID>` to close the issue when the PR is merged -->
+- [ ] Link any dependent pull request in the description
+
+If this PR adds a new feature or fixes a bug:
+
+- [ ] I have added or updated relevant tests
+
```

Comment:
Note: the package will likely updated by `Dependabot`

Reply:
Otherwise, not everyone can test the change on all platform.

I think we should also add a comment 

```markdown
<!-- If you don't have a specific platform, seek help of someone that have it. -->
```

Reply:
:+1: for the suggestions